### PR TITLE
DOGE-66 - Prod pipeline

### PR DESCRIPTION
**Pull request by duraz0rz**
This adds the functional test and production deploy stages to the Jenkins pipeline. Additionally, this also cleans the workspace before doing the build so we don't have lingering files every build.

### CODE REVIEW

File: terraform/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+esac
+
+pushd "providers/aws/us_east_1_$APP_ENVIRONMENT"
+terraform remote config \
+    -backend=S3 \
+    -backend-config="bucket=doge-application" \
+    -backend-config="key=terraform/${APP_ENVIRONMENT}-env-terraform.tfstate" \
+    -backend-config="region=us-east-1"
+
+terraform plan -var ami_id=$AMI_ID
```
Comment (ztbrown) [inline terraform/deploy.sh:34]: is there any case where we might want to plan and apply as separate steps?

File: Jenkinsfile [hunk 2]
```diff
@@ -9,45 +9,94 @@ def getAMIFromPackerManifest() {
 }
 
 pipeline {
-    agent any
+    agent none
```
Comment (ztbrown) [inline Jenkinsfile:12]: Consider `agent any` here and `agent none` at the "wait for deploy to prod" stage.

File: terraform/providers/aws/us_east_1_prod/prod-env.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+variable "ami_id" {}
+
```
Comment (phasebash) [inline terraform/providers/aws/us_east_1_prod/prod-env.tf]: which task service are we pointed towards?

Comment (duraz0rz) [reply terraform/providers/aws/us_east_1_prod/prod-env.tf]: Oops...localhost, apparently (set by `application.properties`). Do we want to stand up a different prod instance? I think, right now, dev points to an existing instance.

**Review by ztbrown — APPROVED** [review]

**Pull request merged**